--- lttng-ust-2.9.0/liblttng-ust-fd/Makefile.am.omv~	2017-06-10 01:30:43.016447988 +0200
+++ lttng-ust-2.9.0/liblttng-ust-fd/Makefile.am	2017-06-10 01:32:26.425966474 +0200
@@ -5,7 +5,7 @@ lib_LTLIBRARIES = liblttng-ust-fd.la
 liblttng_ust_fd_la_SOURCES = \
 	lttng-ust-fd.c
 liblttng_ust_fd_la_LIBADD = \
-	$(top_builddir)/liblttng-ust-comm/liblttng-ust-comm.la
+	$(top_builddir)/liblttng-ust-comm/liblttng-ust-comm.la $(top_builddir)/snprintf/libustsnprintf.la $(top_builddir)/liblttng-ust/liblttng-ust-support.la
 
 if LTTNG_UST_BUILD_WITH_LIBDL
 liblttng_ust_fd_la_LIBADD += -ldl
--- lttng-ust-2.9.0/liblttng-ust/Makefile.am.omv~	2017-06-10 01:21:17.611029749 +0200
+++ lttng-ust-2.9.0/liblttng-ust/Makefile.am	2017-06-10 01:21:25.566838395 +0200
@@ -14,7 +14,7 @@ liblttng_ust_tracepoint_la_SOURCES = \
 	error.h
 liblttng_ust_tracepoint_la_LIBADD = \
 	-lurcu-bp \
-	-lpthread \
+	-lpthread -ldl \
 	$(top_builddir)/snprintf/libustsnprintf.la
 liblttng_ust_tracepoint_la_LDFLAGS = -no-undefined -version-info $(LTTNG_UST_LIBRARY_VERSION)
 liblttng_ust_tracepoint_la_CFLAGS = -DUST_COMPONENT="liblttng_ust_tracepoint" -fno-strict-aliasing
