diff -Naur lttng-ust-2.9.1/liblttng-ust/Makefile.am lttng-ust-2.9.1.tpg/liblttng-ust/Makefile.am
--- lttng-ust-2.9.1/liblttng-ust/Makefile.am	2017-06-12 22:20:03.029190208 +0000
+++ lttng-ust-2.9.1.tpg/liblttng-ust/Makefile.am	2017-10-22 17:38:15.502515305 +0000
@@ -15,6 +15,7 @@
 liblttng_ust_tracepoint_la_LIBADD = \
 	-lurcu-bp \
 	-lpthread \
+	-ldl \
 	$(top_builddir)/snprintf/libustsnprintf.la
 liblttng_ust_tracepoint_la_LDFLAGS = -no-undefined -version-info $(LTTNG_UST_LIBRARY_VERSION)
 liblttng_ust_tracepoint_la_CFLAGS = -DUST_COMPONENT="liblttng_ust_tracepoint" $(AM_CFLAGS)
diff -Naur lttng-ust-2.9.1/liblttng-ust-fd/Makefile.am lttng-ust-2.9.1.tpg/liblttng-ust-fd/Makefile.am
--- lttng-ust-2.9.1/liblttng-ust-fd/Makefile.am	2017-06-12 20:58:38.029190208 +0000
+++ lttng-ust-2.9.1.tpg/liblttng-ust-fd/Makefile.am	2017-10-22 17:37:51.581756529 +0000
@@ -5,7 +5,9 @@
 liblttng_ust_fd_la_SOURCES = \
 	lttng-ust-fd.c
 liblttng_ust_fd_la_LIBADD = \
-	$(top_builddir)/liblttng-ust-comm/liblttng-ust-comm.la
+	$(top_builddir)/liblttng-ust-comm/liblttng-ust-comm.la \
+	$(top_builddir)/snprintf/libustsnprintf.la \
+	$(top_builddir)/liblttng-ust/liblttng-ust-support.la
 
 if LTTNG_UST_BUILD_WITH_LIBDL
 liblttng_ust_fd_la_LIBADD += -ldl
